File: src/test/java/org/rock/paper/scissors/RockPaperScissorsWorkerTest.java
package org.rock.paper.scissors;

import org.junit.Test;
import org.rock.paper.scissors.mock.MockBuilder;

import static org.junit.Assert.assertEquals;

public class RockPaperScissorsWorkerTest {

    private final static String LOSE = "You lose!";
    private final static String WIN = "You win!";
    private final static String TIED = "Match tied!";

    @Test
    public void startRpsWithRockScissorsSelection() {
        MockBuilder.setMockSelection("mockedValue", "r");

        String result = getResult("s");

        assertEquals(result, LOSE);
    }

    @Test
    public void startRpsWithPaperScissorsSelection() {
        MockBuilder.setMockSelection("mockedValue", "p");

        String result = getResult("s");

        assertEquals(result, WIN);
    }

    @Test
    public void startRpsWithRockPaperSelection() {
        MockBuilder.setMockSelection("mockedValue", "r");

        String result = getResult("p");

        assertEquals(result, WIN);
    }

    @Test
    public void startRpsWithScissorsPaperSelection() {
        MockBuilder.setMockSelection("mockedValue", "s");

        String result = getResult("p");

        assertEquals(result, LOSE);
    }

    private String getResult(String yourSelection) {
        RockPaperScissorsWorker worker = new RockPaperScissorsWorker();
        return worker.startRps(null, yourSelection);
    }
}

File: src/test/java/org/rock/paper/scissors/RockPaperScissorsWorkerTest.kt
package org.rock.paper.scissors

import org.junit.Assert
import org.junit.Test
import org.rock.paper.scissors.mock.MockBuilder

class RockPaperScissorsWorkerTest {
    @Test
    fun matchTied() {
        MockBuilder.setMockSelection("mockedValue", "r")

        var result = getResult("r")

        Assert.assertEquals(result, TIED)

        MockBuilder.setMockSelection("mockedValue", "p")

        result = getResult("p")

        Assert.assertEquals(result, TIED)

        MockBuilder.setMockSelection("mockedValue", "s")

        result = getResult("s")

        Assert.assertEquals(result, TIED)
    }

    @Test
    fun startRpsWithRockScissorsSelection() {
        MockBuilder.setMockSelection("mockedValue", "r")

        val result = getResult("s")

        Assert.assertEquals(result, LOSE)
    }

    @Test
    fun startRpsWithPaperScissorsSelection() {
        MockBuilder.setMockSelection("mockedValue", "p")

        val result = getResult("s")

        Assert.assertEquals(result, WIN)
    }

    @Test
    fun startRpsWithRockPaperSelection() {
        MockBuilder.setMockSelection("mockedValue", "r")

        val result = getResult("p")

        Assert.assertEquals(result, WIN)
    }

    @Test
    fun startRpsWithScissorsPaperSelection() {
        MockBuilder.setMockSelection("mockedValue", "s")

        val result = getResult("p")

        Assert.assertEquals(result, LOSE)
    }

    @Test
    fun startRpsWithPaperRockSelection() {
        MockBuilder.setMockSelection("mockedValue", "p")

        val result = getResult("r")

        Assert.assertEquals(result, LOSE)
    }

    @Test
    fun startRpsWithScissorsRockSelection() {
        MockBuilder.setMockSelection("mockedValue", "s")

        val result = getResult("r")

        Assert.assertEquals(result, WIN)
    }

    private fun getResult(yourSelection: String): String {
        return RockPaperScissorsWorker().startRps(null, yourSelection)
    }

    companion object {
        private const val LOSE = "You lose!"
        private const val WIN = "You win!"
        private const val TIED = "Match tied!"
    }
}